connectivity/check/action.go

```diff
@@ -163,23 +167,24 @@ func (a *Action) ExecInPod(ctx context.Context, cmd []string) {
 	cmdName := cmd[0]
 	cmdStr := strings.Join(cmd, " ")
 
-	if stderr.Len() > 0 {
-		a.test.Debugf("%s stderr: %s", cmdName, stderr.String())
-	} else if stdout.Len() > 0 {
-		a.test.Debugf("%s stdout: %s", cmdName, stdout.String())
-	}
-
 	if err != nil || stderr.Len() > 0 {
 		if a.shouldSucceed() {
 			a.Failf("command %q failed: %s", cmdStr, err)
 		} else {
-			a.test.Debugf("command %q failed as expected: %s", cmdStr, err)
+			a.test.Infof("command %q failed as expected: %s", cmdStr, err)
 		}
 	} else {
 		if !a.shouldSucceed() {
 			a.Failf("command %q succeeded while it should have failed: %s", cmdStr, stdout.String())
 		}
 	}
+	if a.failed {
```

Comment:
maybe better to do:
```suggestion
	if a.failed && a.shouldSucceed() {
```
so that we don't print warnings for expected failures.

Reply:
`a.failed` becomes `true` only if the action failed, i.e., if the test comment failed while it should have succeeded, or if the test command succeeded while it should have failed.